helper/decoder/decoder.go

```diff
@@ -24,20 +24,25 @@ import (
 	"github.com/alibaba/ilogtail/helper/decoder/influxdb"
 	"github.com/alibaba/ilogtail/helper/decoder/opentelemetry"
 	"github.com/alibaba/ilogtail/helper/decoder/prometheus"
+	"github.com/alibaba/ilogtail/helper/decoder/raw"
 	"github.com/alibaba/ilogtail/helper/decoder/sls"
 	"github.com/alibaba/ilogtail/helper/decoder/statsd"
+	"github.com/alibaba/ilogtail/pkg/models"
 	"github.com/alibaba/ilogtail/pkg/protocol"
 )
 
 // Decoder used to parse buffer to sls logs
 type Decoder interface {
 	// Decode reader to logs
 	Decode(data []byte, req *http.Request) (logs []*protocol.Log, err error)
+	// DecodeV2 reader to groupEvents
+	DecodeV2(data []byte, req *http.Request) (groupEvents *models.PipelineGroupEvents, err error)
```

Comment:
一个req里面是不是可能包含多个GroupEvents(比如open telemetry的request就是这个情况)，是不是返回[]*models.PipelineGroupEvents比较好，

Reply:
好的，之前没有考虑到PipelineGroupEvents多个场景，在这个场景下，根据input配置参数QueryParam去设置metadata的时候，会有什么额外的逻辑需要考虑吗？还是就是同一个req生成的多个group的metadata是相同的

Reply:
[fix: DecoderV2 return PipelineGroupEvents array instead of single Pip…](https://github.com/alibaba/ilogtail/pull/599/commits/e573eeda20b7b29b254d01f30f81697063ea2051)  已修改为返回数组